tests/integration/conftest.py

```diff
@@ -662,3 +700,147 @@ def kernel(shell):
 
         # Reboot after clean-up
         shell("tmt-reboot -t 600")
+
+
+@pytest.fixture()
+@pytest.mark.yum_conf_exclude
+def yum_conf_exclude(shell, backup_directory, request):
+    """
+    Fixture.
+    Define `exclude=kernel kernel-core` in /etc/yum.conf.
+    Using pytest.mark.parametrize pass the packages to exclude in a single scenario as a list.
+    The fixture is called indirectly in the test function.
+    Example:
+        # one testcase
+        @pytest.mark.parametrize("yum_conf_exclude", [["this", "that", "also_this"]])
+
+        # more testcases
+        @pytest.mark.parametrize("yum_conf_exclude", [["this", "that"], ["then_this"]])
+
+        def test_function(yum_conf_exclude):
+    """
+    exclude = ["rhn-client*"]
+    if hasattr(request, "param"):
+        exclude = request.param
+        print(">>> Using parametrized packages requested in the fixture.")
+    yum_config = "/etc/yum.conf"
+    backup_dir = os.path.join(backup_directory, "yum")
+    shell(f"mkdir -v {backup_dir}")
+    config_bak = os.path.join(backup_dir, os.path.basename(yum_config))
+    config = configparser.ConfigParser()
+    config.read(yum_config)
+
+    assert shell(f"cp -v {yum_config} {config_bak}").returncode == 0
+
+    for pkg in exclude:
+        # If there is already an `exclude` section, append to the existing value
+        if config.has_option("main", "exclude"):
+            pre_existing_value = config.get("main", "exclude")
+            config.set("main", "exclude", f"{pre_existing_value} {pkg}")
+        else:
+            config.set("main", "exclude", pkg)
```

Comment:
We could rewrite this section as this: 


```suggestion
    pkgs_to_exclude = " ".join(exclude)
    # If there is already an `exclude` section, append to the existing value
    if config.has_option("main", "exclude"):
        pre_existing_value = config.get("main", "exclude")
        config.set("main", "exclude", f"{pre_existing_value} {pkgs_to_exclude}")
    else:
        config.set("main", "exclude", pkgs_to_exclude)
```

Main idea here is to not loop through every exclude package and perform the same operations every time (check for option, get the value, rewrite it, ...) 

We can loop through the packages once and then just append to the config once too

Reply:
Good idea, thank you @r0x0d !